app/src/services/queueSvc.js

```diff
@@ -18,129 +18,165 @@
-   *  @param {object} message - a Message object
-   *  @param {object} opts - Bull opts, including setting delay time
+   * @param {string} client - the client that owns the message
+   * @param {object} message - a Message object
+   * @param {object} opts - Bull opts, including setting delay time
    */
-  async enqueue (client, message, opts = {}) {
-    const job = this.queue.add({
+  async enqueue(client, message, opts = {}) {
+    await this.dataService.updateStatus(client, message.messageId, queueState.ENQUEUED);
+    await this.queue.add({
       client: client,
       messageId: message.messageId
     }, Object.assign(opts, {
       jobId: message.messageId
-    }));
-
-    await this.dataService.updateStatus(client, message.messageId, 'enqueued');
-
-    log.info('enqueue', `Job ${message.messageId} enqueued`);
-    return job.id;
+    }))
+      .then(job => log.info('enqueue', `Job ${job.id} enqueued`))
+      .catch(e => log.error(e));
   }
 
-  /** @function updateContent
-   *  Update the persisted content (email message) for a Message.
-   *  When jobs are completed or failed/errored, we want to remove the email content.
+  /**
+   * @function updateContent
+   * Update the persisted content (email message) for a Message.
+   * When jobs are completed or failed/errored, we want to remove the email content.
    *
-   *  @param {object} job - the queue job
+   * @param {object} job - the queue job
    */
-  async updateContent (job) {
-    if (job && job.data && job.data.messageId && job.data.client) {
-      await this.dataService.deleteMessageEmail(job.data.client, job.data.messageId);
+  async updateContent(job) {
+    if (job && job.data) {
+      if (job.data.messageId && job.data.client) {
+        await this.dataService.deleteMessageEmail(job.data.client, job.data.messageId);
+      }
+      await job.update(null); // Scrub out client and message id
     }
-    await job.update(null); // Scrub out client and message id
+
   }
 
-  /** @function updateStatus
-   *  Update the persisted status for a Message
+  /**
+   * @function updateStatus
+   * Update the persisted status for a Message
    *
-   *  @param {object} job - the queue job
-   *  @param {string} status - the queue related status, (potentiall) different a business status
-   *  @param {string} description - optional description for the status, generally an error message
+   * @param {object} job - the queue job
+   * @param {string} status - the queue related status
+   * @param {string} description - optional description for the status, generally an error message
    */
-  async updateStatus (job, status, description) {
+  async updateStatus(job, status, description) {
     if (job && job.data && job.data.messageId && job.data.client) {
-      await this.dataService.updateStatus(job.data.client, job.data.messageId, status, description);
+      this.dataService.updateStatus(job.data.client, job.data.messageId, status, description);
```

Comment:
I added it back here because there is some value in being able to choose to stall on updateStatus before proceeding. However, in all our use cases right now, we actually have no requirement for awaiting on the queueService.updateStatus as we just need to ensure that a transaction will be dispatched before work continues. By virtue of dataService.updateStatus being a transaction, the DB will block until that gets through or rolled back before proceeding anyways, which will ensure other operations end up happening in the right order there anyways.

Reply:
Counting on the db and pools to enforce our logic flow is not a great idea.  Chances are very slim, but no guarantee when the next pool connection is available or which caller will get to execute their query.

Reply:
Yep - I added back some awaits where it made sense - refer to later comment.